include/boost/gil/channel_algorithm.hpp

```diff
@@ -329,7 +329,7 @@ template <> struct channel_convert_to_unsigned<bits16s> : public std::unary_func
 
 template <> struct channel_convert_to_unsigned<bits32s> : public std::unary_function<bits32s,bits32> {
     typedef bits32 type;
-    type operator()(bits32s x) const { return static_cast<bits32>(x+(1<<31)); }
+    type operator()(bits32s x) const { return static_cast<bits32>(x)+(static_cast<bits32>(1)<<31); }
 };
```

Comment:
I would prefer `1u` over `static_cast<bits32>(1)`, just for conciseness, but that's really nit-picky now.
Other than that, the PR looks good to me, so I believe we should merge it, even though I don't see a specific test failure in our CI runs that this fixes.

Reply:
Done. Commits squashed. PR updated.

> I don't see a specific test failure in our CI runs that this fixes.

I thought that this actually is related to #51 and fixes GCC build https://travis-ci.org/boostorg/gil/jobs/355009262